lib/spack/spack/environment.py

```diff
@@ -122,6 +122,10 @@ def execute(self):
                        if x != os.path.normpath(self.value)]
         os.environ[self.name] = self.separator.join(directories)
 
+class PushEnv(NameValueModifier):
```

Comment:
I don't think `PushEnv` belongs to the commands here, as it is specific to `lmod` (while `EnvironmentModifications` represents a *generic* list of commands to modify the environment).